tensorly/preprocessing.py

```diff
@@ -0,0 +1,132 @@
+    tensor slice and fitting the PARAFAC2 model to the scores. Then, to get back the components,
+    we left-multiply the :math:`B_i`-matrices with the loading matrices.
+
+    [1]_ states that we can constrain our :math:`B_i`-matrices to lie in a given vector space,
+    :math:`\mathscr{V}_i` by multiplying the data matrices with an orthogonal basis matrix that
+    spans :math:`\mathscr{V}_i`. However, since we know that :math:`B_i` lie in the column space
+    of :math:`X_i`, we can multiply the :math:`X_i`-matrices by an orthogonal matrix that spans
+    :math:`\text{col}(X_i)` without affecting the fit of the model. Thus we can compress our data
+    prior to fitting the PARAFAC2 model whenever the number of rows in our data matrices exceeds
+    the number of columns (as the rank of :math:`\text{col}(X_i)` cannot exceed the number of rows).
+
+    To implement this, we use the SVD to get an orthogonal basis for the column space of :math:`X_i`.
+    Moreover, since :math:`S_i V_i^T = U_i^T X_i`, we can skip an additional matrix multiplication
+    by fitting the model to :math:`S_i V_i^T`.
+
+    Finally, we note that this approach can also be implemented by truncating the SVD. If an appropriate
+    threshold is set, this will not affect the fitted model in any major form.
+
+    .. note::
+        This can be thought of as a simplified version of the DPAR approach for compressing PARAFAC2 models [2]_,
+        which compresses all modes of :math:`\mathcal{X}` to fit an approximate PARAFAC2 model.
+
+    Parameters
+    ----------
+    tensor_slices : list of matrices
+        The data matrices to compress
+    compression_threshold : float (0 <= compression_threshold <= 1)
+        Threshold at which the singular values should be truncated. Any singular value less than
+        compression_threshold * s[0] is set to zero. Note that if this is nonzero, then the found
+        components will likely be affected.
+    svd : str, default is 'truncated_svd'
+        function to use to compute the SVD, acceptable values in tensorly.SVD_FUNS
+
+    Returns
+    -------
+    list of matrices
+        The score matrices, used to fit the PARAFAC2 model to.
+    list of matrices
+        The loading matrices, used to decompress the PARAFAC2 components after fitting
+        to the scores.
+
+    References
+    ----------
+    .. [1] Helwig, N. E. (2017). Estimating latent trends in multivariate longitudinal
+           data via Parafac2 with functional and structural constraints. Biometrical
+           Journal, 59(4), 783-803. doi: 10.1002/bimj.201600045
+
+    .. [2] Jang JG, Kang U. Dpar2: Fast and scalable parafac2 decomposition for
+           irregular dense tensors. In2022 IEEE 38th International Conference on Data
+           Engineering (ICDE) 2022 May 9 (pp. 2454-2467). IEEE.
+
+    """
+    loading_matrices = [None for _ in tensor_slices]
+    score_matrices = [None for _ in tensor_slices]
```

Comment:
I like to use `X.append()` rather than pre-filling the list, but this is just a matter of style.

Reply:
We arbitrarily landed at this because we use None as default value if we're not compressing the data. Also, this way we'd get an error downstream if we for some reason forgot to add a score matrix (we would get a type error instead of just a slightly smaller tensor).

Reply:
Right. Now understanding how `None` is handled later, this seems like a good strategy.